### ISSUE TITLE
Probem with PWM pairs

### ISSUE THREAD

**Issue by aleandros** [adafruit/adafruit-beaglebone-io-python#66]
I get a "RuntimeError: You must start() the PWM channel first" whenever I use two "contiguous" PWM pins. For example, If I want to use P9_29 and P9_31, which correspond to EHRPWM0A and EHRPWM0B (or P9_14 and P9_16), I can start them with PWM.start, and I can change the duty cycle of the first one without problems, but when I try to call PWM.set_duty_cycle on the second one, I get the aforementioned error.

Thanks in advance!

**Comment by JoshKeaper** [comment]
The problem is still there. Sometimes it happens, sometimes not. Start-stop-cleanup-start cycles occasionally help, but more often than not, the only way to get all PWMs up is to reboot and try again.

adafruit-beaglebone-io-python 0.0.30 (previously used 0.0.19, same problem)
Linux version 3.8.13-bone71 (root@a3-imx6q-wandboard-2gb) (gcc version 4.6.3 (Debian 4.6.3-14) ) #1 SMP Tue Mar 17 18:07:44 UTC 2015
Debian GNU/Linux 7.8 (wheezy)
BeagleBoard.org BeagleBone Debian Image 2014-04-23
BeagleBone Black Rev C (also tried Rev B, same problem)

**Comment by KarthikNayak** [comment]
Bumping this problem, still needs to be fixed.

**Comment by raunaqbhirangi** [comment]
Any updates on this issue? I just can't get around this.

**Comment by JoshKeaper** [comment]
[These instructions worked for me.](https://randomlinuxhacks.wordpress.com/2014/09/16/changing-pwm-period-on-beaglebone-black/) Note that this includes applying `ti_ehr_pwm.patch` and building the 3.8-rt kernel. It seems to work for some kernel versions (Debian 7.5, `3.8.13-bone50`), but fails for later versions (Debian 7.8, `3.8.13-bone70`). I've also tried [this solution](http://saadahmad.ca/using-pwm-on-the-beaglebone-black/), but it doesn't work without patching the kernel.
These instructions definitely won't work for 4.* kernels, since they've redesigned cape manager, device tree compiler and overlays. [More info here.](https://groups.google.com/forum/#!category-topic/beagleboard/support/dkS51WbicTo)
I'll probably stick with 3.8 until there are clear solutions for later kernels.

**Comment by raunaqbhirangi** [comment]
The best and simplest solution to resolve this problem is running it on the default frequency of 2kHz. This seems to work.

**Comment by JoshKeaper** [comment]
Go ahead and drive some servos with 2 kHz.

**Comment by pdp7** [comment]
It looks like this commit may have addressed the issue in the kernel:
```
commit 2543b5f335e2812369d0de2d21dc92526832b7d5
Author: Robert Nelson <robertcnelson@gmail.com>
Date:   Tue Sep 15 14:26:18 2015 -0500

    pwm: merge https://github.com/beagleboard/linux/pull/45
    
    Signed-off-by: Robert Nelson <robertcnelson@gmail.com>

 patch.sh                                    |  4 +++-
 patches/fixes/0011-PWM-period-control.patch | 81 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 patches/fixes/0012-PWM-period-control.patch | 53 +++++++++++++++++++++++++++++++++++++++++++
 version.sh                                  |  2 +-
 4 files changed, 138 insertions(+), 2 deletions(-)
```

The Python example above does seem to be working in 3.8.13-bone80:
* Repo: http://github.com/RobertCNelson/bb-kernel
* Branch: am33x-v3.8

**Comment by pdp7** [comment]
@aleandros  There have been several PWM improvements from merged PRs, most recently #108. Please try again and report if this is still an issue.
